src/server/game/Entities/Creature/Creature.h

```diff
@@ -222,6 +223,7 @@ class Creature : public Unit, public GridObject<Creature>, public MovableMapObje
     bool CanStartAttack(Unit const* u) const;
     float GetAggroRange(Unit const* target) const;
     float GetAttackDistance(Unit const* player) const;
+    float GetDetectionRange() const { return m_detectionDistance; }
```

Comment:
please consider `[[nodiscard]]` where applicable

Reply:
You are right, the return function is not to be discarded

Reply:
fixed